### PR TITLE
Working prototype on launching missiles without cancelling orders

### PR DESCRIPTION
**Pull request by Garanas**
As requested by the balance team, a prototype of the feature. It isn't complete yet, needs some additional checks before firing the weapon.

https://user-images.githubusercontent.com/15778155/185418565-906f553a-206a-4d23-9b3c-d5f9c3ded5e1.mp4

### CODE REVIEW

File: lua/SimCallbacks.lua [hunk 1]
```diff
@@ -727,6 +727,48 @@ do
...
+                if blueprint.ManualFire and blueprint.CountedProjectile then
+
+                    if blueprint.EnabledByEnhancement then
+                        local enhancements = SimUnitEnhancements[unit.EntityId]
+                        if enhancements then
+                            for _, enh in enhancements do
+                                if enh == blueprint.EnabledByEnhancement then
+                                    LOG("hellooo")
+                                    weapon = unit:GetWeaponByLabel(blueprint.Label)
+                                    break
```
Comment (Hdt80bro) [inline lua/SimCallbacks.lua:747]: This only breaks out of the inner loop, not the outer one.

And so people argue that this is the only place `goto`s have in languages without named breaks & continues: to break out of outer loops.

Unfortunately, we don't get those in Lua 5.0, but a `return` can do the same thing if the code is moved to a new function (which is usually what should have already been done if there's that much nesting and indentation).

Comment (Garanas) [reply lua/SimCallbacks.lua:747]: Ah, whoops 😄 !

I'll fix that, but the concept works regardless!

**Comment by lL1l1** [comment]
Would be interesting to apply to overcharge if possible.

**Comment by Garanas** [comment]
Yes, it would work in a similar fashion.

**Comment by MrRowey** [comment]
@Garanas any idea if this pr will be completed?

**Comment by MrRowey** [comment]
@Garanas Do we see this becoming a feature at any point or would it just be safe to close this Prs for now as it sat dormant for over 2 years

**Comment by MrRowey** [comment]
I am going to close this PR as there has been no work done with it since 2022. It can be re-opened if someone is going to take it on and complete the idea

**Pull request closed without merging**